### PR TITLE
Add or suppress missing return type declarations

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

File: src/Controller/Controller.php [hunk 1]
```diff
@@ -292,7 +292,7 @@ public function loadComponent(string $name, array $config = []): Component
      * @param string $name Property name
      * @return \Cake\Controller\Component|\Cake\ORM\Table|null
      */
-    public function __get(string $name)
+    public function __get(string $name): Component|Table|null
```
Comment (ADmad) [inline src/Controller/Controller.php]: Since return types cannot be widened perhaps we should leave out the types for magic methods?

**Comment by othercorey** [comment]
I'll backport the closure changes for easier merging.

File: src/Controller/Controller.php (hunk 1 above)
Comment (othercorey) [reply src/Controller/Controller.php]: Should we set it to `mixed` to so a return value is required since no type declaration is `mixed|void`?

Comment (othercorey) [reply src/Controller/Controller.php]: It seems that is the function signature from php docs:
> public __get(string $name): mixed

Comment (ADmad) [reply src/Controller/Controller.php]: I guess we should add `mixed` then.

**Review by ADmad — APPROVED** [review]

**Pull request merged**